tests/test_mq.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2018 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation; either version 2 of the License, or (at your option) any later
+# version.
+#
+# REANA is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with REANA; if not, see <http://www.gnu.org/licenses>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization or
+# submit itself to any jurisdiction.
+
+"""REANA-Commons message queue publisher tests."""
+
+import json
+import threading
+from unittest.mock import ANY, patch
+
+import pytest
+from kombu import Connection, Exchange, Queue
+from kombu.exceptions import OperationalError
+
+from reana_commons.publisher import WorkflowStatusPublisher
+
+
+def test_consume_msg(ConsumerBaseOnMessageMock, in_memory_queue_connection,
+                     default_queue, default_in_memory_producer,
+                     consume_queue):
+    """Test message is consumed from the queue."""
+    consumer = ConsumerBaseOnMessageMock(
+        connection=in_memory_queue_connection,
```

Comment:
Probably the `ConsumerBase` should be `InMemoryConsumer` and be already instantiated, but since in the `test_server_unreachable` I want to provide a custom connection I did it this way...